spring-cloud-zookeeper-core/src/main/java/org/springframework/cloud/zookeeper/ZookeeperAutoConfiguration.java

```diff
@@ -56,18 +59,23 @@ public ZookeeperProperties zookeeperProperties() {
 	public CuratorFramework curatorFramework(
 			RetryPolicy retryPolicy,
 			ZookeeperProperties properties,
+			List<CuratorFrameworkCustomizer> customizerList,
 			ObjectProvider<EnsembleProvider> optionalEnsembleProvider,
 			ObjectProvider<TracerDriver> optionalTracerDriverProvider)
 	throws Exception {
 		CuratorFrameworkFactory.Builder builder = CuratorFrameworkFactory.builder();
+		if (!CollectionUtils.isEmpty(customizerList))	{
+			customizerList.forEach(curatorFrameworkCustomizer -> curatorFrameworkCustomizer.customize(builder));
+		}
```

Comment:
Then call `customizerList.orderdStream()` which will automatically apply spring ordering to customizers. This should also probably move to the end so that user can override any things done to the builder here (though unlikely).